R/zchunk_LB132.ag_an_For_Prices_USA_C_2005.R

```diff
@@ -38,43 +39,237 @@ module_aglu_LB132.ag_an_For_Prices_USA_C_2005_DISABLED <- function(command, ...)
+      group_by(countries, item) %>%
+      summarise_at(vars(prod), mean, na.rm = TRUE) %>%
+      ungroup() %>%
+      filter(!is.na(prod)) %>%
+      # Match production and price for each FAO item to calculate revenue, avoid any missing value by inner_join
+      inner_join(Avg_price, by = c("countries", "item")) %>%
+      # Calculate revenue by commodity as production times price
+      mutate(V_USD = price * prod) %>%
+      left_join(FAO_an_items_PRODSTAT, by = "item") %>%
+      filter(!is.na(GCAM_commodity)) %>%
+      # Aggregate revenue and production by GCAM commodity
+      group_by(GCAM_commodity) %>%
+      summarize_at(vars(V_USD, prod), sum, na.rm = TRUE) %>%
+      ungroup() %>%
+      # Calculate production weighted average price for each GCAM commodity
+      mutate(Price_USDt = V_USD / prod) %>%
+      # Convert to model units
+      mutate(calPrice = round(Price_USDt * CONV_T_METRIC_SHORT * CONV_2004_1975_USD / CONV_T_KG, digits = DIGITS_CALPRICE)) %>%
+      select(GCAM_commodity, calPrice) ->
+      Price_an
+
+    # Part 2: Fodder crops and pasture
+    # Calculate average FodderHerb prices from alfalfa prices
+    USDA_Alfalfa_prices_USDt %>%
+      select(year, avg) %>%
+      # Calculate average Alfalfa price over the price years
+      filter(year %in% MODEL_PRICE_YEARS) %>%
+      summarise_at(vars(avg), mean, na.rm = TRUE) %>%
+      rename(FodderHerb = avg) %>%
+      # Note: Setting FodderGrass price as a ratio to FodderHerb
+      mutate(FodderGrass = FodderHerb * PRICERATIO_GRASS_ALFALFA) %>%
+      # NOTE: Setting Pasture price equal to FodderGrass price
+      mutate(Pasture = FodderGrass) %>%
+      gather(GCAM_commodity, Price_USDt) %>%
+      # Combine crop commodities to get all primary agricultural commodities in a single tibble
+      bind_rows(Price_ag) %>%
+      # Convert to model units
+      mutate(calPrice = round(Price_USDt * CONV_2004_1975_USD / CONV_T_KG, digits = DIGITS_CALPRICE)) %>%
+      select(-Price_USDt) %>%
+      # Combine animal products
+      bind_rows(Price_an) %>%
+      mutate(unit = "1975$/kg") ->
+      Price_ag_an
+
+    # Part 3: Forest products, use FAO forest export value and export quantities to estimate price
+    FAO_USA_For_Exp_t_USD_FORESTAT %>%
+      select(-countries, -country.codes, -item, -item.codes, -element.codes) %>%
+      gather(year, value, -element) %>%
+      # Calculate average value and quantity over price years
+      filter(year %in% MODEL_PRICE_YEARS) %>%
+      group_by(element) %>%
+      summarise_at(vars(value), mean, na.rm = TRUE) %>%
+      ungroup() %>%
+      # Modity element names to one word so that they can be used as column names when spreading element and doing calculations
+      mutate(element = if_else(element == "Export Quantity (m3)", "Exp_m3", "ExpV_USD")) %>%
+      mutate(GCAM_commodity = "Forest") %>%
+      spread(element, value) %>%
+      # Calculate forest price as export value (in thous USD) divided by export quantity
```

Comment:
Unfortunately roundwood producer prices aren't reported in FAOSTAT, so our "Forest" commodity prices are calculated as total export value divided by total export quantity. Obviously not ideal, and there's a chance that FAOSTAT will start reporting producer prices, so something we should keep an eye on.